scripts/update-development-status.py

```diff
@@ -0,0 +1,112 @@
+"""
+This script updates games development status
+
+To run, install from pip:
+- pygithub
+- python-gitlab
+
+Add environment variables:
+- GH_TOKEN (see https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-token)
+- GL_TOKEN (see https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token)
+"""
+import os
+import re
+import yaml
+
+from pathlib import Path
+from datetime import datetime, timedelta
+from github import Github, GithubException
+from gitlab import Gitlab
+
+GH_REGEX = re.compile(r'https://github.com/([^/]+)/([^/]+)')
+GL_REGEX = re.compile(r'https://gitlab.com/([^/]+/[^/]+)')
+
+
+def main():
+    gh = Github(os.environ["GH_TOKEN"])
+    gl = Gitlab('https://gitlab.com', private_token=os.environ["GL_TOKEN"])
+
+    for filename in Path('games').iterdir():
+        if not filename.is_file() or filename.suffix != '.yaml':
+            continue
+
+        games = yaml.safe_load(open(filename, encoding='utf-8'))
+        for game in games:
+            repo_url = game.get('repo', '')
+
+            if game.get('development', '') == 'complete':
+                continue
+
+            latest_commit_date = get_latest_commit_date(repo_url, gh, gl)
+
+            if latest_commit_date is None:
+                continue
+
+            diff = datetime.now() - latest_commit_date
+
+            if diff < timedelta(weeks=1):
+                game['development'] = 'very active'
+            elif diff < timedelta(weeks=4):
+                game['development'] = 'active'
+            elif diff < timedelta(days=365):
+                game['development'] = 'sporadic'
+            else:
+                game['development'] = 'halted'
+
+        yaml.dump(games, open(filename, 'w', encoding='utf-8'), sort_keys=False)
+        print(filename, 'has been updated')
+
+
+def is_github_repo(repo):
+    return 'https://github.' in repo
+
+
+def is_gitlab_repo(repo):
+    return 'https://gitlab.' in repo
```

Comment:
should check that the full host is gitlab.com
Some repos use a self-hosted gitlab; the GL_TOKEN isn't going to work there

Reply:
Updated